### ISSUE TITLE
Same prediction every time.

### ISSUE THREAD

**Issue by dberma15** [fchollet/keras#7023]
Please make sure that the boxes below are checked before you submit your issue. If your issue is an implementation question, please ask your question on [StackOverflow](http://stackoverflow.com/questions/tagged/keras) or [join the Keras Slack channel](https://keras-slack-autojoin.herokuapp.com/) and ask there instead of filing a GitHub issue.

Thank you!

- [ x] Check that you are up-to-date with the master branch of Keras. You can update with:
pip install git+git://github.com/fchollet/keras.git --upgrade --no-deps

- [ x] If running on TensorFlow, check that you are up-to-date with the latest version. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [ ] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps

- [ ] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

Hi,

I’m trying to do an image classifier that determines if something should be given a specific hashtag or not. My problem is that the accuracy of the classifier after each epoch remains constant, and is essentially assigning the same class to all images. This makes no sense as the image classes are fairly distinct (#gym and #foraging). I basically copied the smaller CNN you used:

model = Sequential()
model.add(Convolution2D(32, 3, 3, input_shape=(3, 100, 100), activation=’relu’))
model.add(Dropout(0.2))
model.add(Convolution2D(32, 3, 3, activation=’relu’))
model.add(MaxPooling2D(pool_size=(2, 2)))
model.add(Flatten())
model.add(Dense(512, activation=’relu’))
model.add(Dropout(0.5))
model.add(Dense(1, activation=’softmax’))

epochs = 10
lrate = 0.001
decay = lrate/epochs
sgd = SGD(lr=lrate, momentum=0.9, decay=decay, nesterov=False)
print(‘compiling model’)
model.compile(loss=’binary_crossentropy’, optimizer=sgd, metrics=[‘accuracy’])

print(“fitting model”)
model.fit(X_training, Y_training, nb_epoch=epochs, batch_size=100)

But every time, I get these results:

fitting model
Train on 908 samples, validate on 908 samples
Epoch 1/10
908/908 [==============================] – 104s – loss: 7.9712 – acc: 0.5000
Epoch 2/10
908/908 [==============================] – 104s – loss: 7.9712 – acc: 0.5000
Epoch 3/10
908/908 [==============================] – 104s – loss: 7.9712 – acc: 0.5000
Epoch 4/10
908/908 [==============================] – 104s – loss: 7.9712 – acc: 0.5000
Epoch 5/10
908/908 [==============================] – 104s – loss: 7.9712 – acc: 0.5000
Epoch 6/10
908/908 [==============================] – 104s – loss: 7.9712 – acc: 0.5000
Epoch 7/10
908/908 [==============================] – 104s – loss: 7.9712 – acc: 0.5000
Epoch 8/10
908/908 [==============================] – 104s – loss: 7.9712 – acc: 0.5000
Epoch 9/10
908/908 [==============================] – 104s – loss: 7.9712 – acc: 0.5000
Epoch 10/10
908/908 [==============================] – 104s – loss: 7.9712 – acc: 0.5000

What am I doing wrong?

**Comment by danielS91** [comment]
A softmax layer with one neuron makes no sense, since of the softmax function will always be 1.0. If you want to do a binary classification you either have to use a one one neuron with sigmoid activation neuron or two neurons with softmax activation.

**Comment by dberma15** [comment]
Unfortunately, that didn't really help. The accuracy is still the same on each iteration epoch for the validation data set, and is always around 0.45.

**Comment by dattarajrao** [comment]
i had the same issue and this got fixed by changing "Sigmoid" for the 1 Neuron Dense Layer to "Softmax".
